### ISSUE TITLE
(unnecessary?) sync locking when checking if Native is loaded causes unneeded slowdown

### ISSUE THREAD

**Issue by noamBarkai** [jpountz/lz4-java#80]
When using this library in a highly concurrent setting - vis. Apache Spark - I'm seeing a lot of contention around [this](https://github.com/jpountz/lz4-java/blob/master/src/java/net/jpountz/util/Native.java#L66]) synchronisation block, specifically when called from [here](https://github.com/jpountz/lz4-java/blob/master/src/java/net/jpountz/lz4/LZ4Factory.java#L135)
It seems the synchronisation is somewhat redundant and that it can safely be removed, assuming the boolean member ```loaded``` is modified to volatile, no?
I can offer a PR if you feel like wise.

**Comment by odaira** [comment]
Yes, I think changing `loaded` to volatile will work.  If you still see the contention problem and if you submit a PR, I'll take a look at it.

**Comment by noamBarkai** [comment]
@odaira - it's been a while since this issue was opened... :)
I'm not sure if this problem still persists, but regardless I can open a PR, no problem, I don't think it's harmful.

**Comment by noamBarkai** [comment]
https://github.com/lz4/lz4-java/pull/126

**Comment by odaira** [comment]
Thanks much for your contribution. Because I am about to release a new version, I'll review this after that.

**Comment by JoshRosen** [comment]
In Spark, https://github.com/apache/spark/pull/24905 may partially mitigate this by significantly reducing the number of calls to `fastestInstance()`.
